cimgraph/models/distributed_area.py

```diff
@@ -19,542 +19,42 @@ def __post_init__(self):
         self.distributed_areas = []
         self.boundaries = []
 
-    def build_from_area(self):
+    def build_from_area(self) -> None:
         self.get_all_edges(self.container.__class__)
         self.connection.create_distributed_graph(area=self.container, graph=self.graph)
 
 
+    # Build the distributed model from a topology message from GridAPPS-D
+    def build_from_topo_message(self, topology_dict:dict):
+        # Set base as
+        self.container = self.add_jsonld_to_graph(topology_dict)
 
-    # def build_from_topo_message(self, topology_dict: dict):
+        # builder for Feeder, Substation, VoltageLevel, and Bay
+        if isinstance(self.container, self.cim.EquipmentContainer):
+            for addr_eq in topology_dict['AddressableEquipment']:
+                equipment = self.add_jsonld_to_graph(addr_eq)
+                equipment.EquipmentContainer = self.container
 
+            for unaddr_eq in topology_dict['AddressableEquipment']:
+                equipment = self.add_jsonld_to_graph(unaddr_eq)
+                equipment.EquipmentContainer = self.container
 
+            for meas in topology_dict['Measurements']:
+                self.add_jsonld_to_graph(meas)
 
 
-#         AddrEquip = AddressableEquipment(self.cim)
-#         UnaddrEquip = UnaddressableEquipment(self.cim)
-#         if centralized_graph:
-#             for node_mrid in topology_dict['connectivity_node']:
-#                 try:
-#                     uuid = UUID(node_mrid.strip('_').lower())
-#                     node = centralized_graph[self.cim.ConnectivityNode][uuid]
-#                     self.add_to_graph(node)
-#                     for terminal in node.Terminals:
-#                         self.add_to_graph(terminal)
-#                         self.add_to_graph(terminal.ConductingEquipment)
-#                 except:
-#                     _log.warning('node ' + str(uuid) + ' not in feeder')
+        elif isinstance(self.container, self.cim.SubSchedulingArea):
+            for boundary in topology_dict['BoundaryTerminals']:
```

Comment:
This should be:
for boundary in topology_dict['FeederArea']['BoundaryTerminals']: